src/windows/pdb.rs

```diff
@@ -305,12 +306,20 @@ impl PDBInfo<'_> {
         let pi = pdb.pdb_information()?;
         let dbi = pdb.debug_information()?;
         let frame_table = pdb.frame_table()?;
-
         let cpu = Self::get_cpu(&dbi);
         let debug_id = Self::get_debug_id(&dbi, pi);
+        let pe_debug_id = get_pe_debug_id(pe.as_ref());
+
+        if let Some(pe_debug_id) = pe_debug_id {
+            if debug_id != pe_debug_id {
+                return Err(From::from(format!(
+                    "Debug ids aren't the same: {} (pdb) != {} (pe)",
+                    debug_id, pe_debug_id
+                )));
+            }
+        }
```

Comment:
This change is unrelated to the rest, split it out in a separate commit.This change is unrelated to the rest, split it out in a separate commit. It would also be nice to have a test for it.